src/v/archival/ntp_archiver_service.cc

```diff
@@ -755,6 +756,50 @@ ss::future<bool> ntp_archiver::maybe_upload_manifest() {
     }
 }
 
+ss::future<> ntp_archiver::maybe_flush_delta_offset_next() {
+    auto& manifest = _parent.archival_meta_stm()->manifest();
+    if (manifest.delta_offset_next() != model::offset_delta{}) {
+        co_return;
+    }
+    if (manifest.get_last_offset() == model::offset{}) {
+        co_return;
+    }
+    auto sync_timeout = config::shard_local_cfg()
+                          .cloud_storage_metadata_sync_timeout_ms.value();
+    bool is_synced = co_await _parent.archival_meta_stm()->sync(sync_timeout);
+    if (!is_synced) {
+        vlog(
+          _rtclog.warn,
+          "Failed to sync before replicating delta_offset_next update for "
+          "archival_metadata_stm");
+    }
+    // Translate just past the end, and persist the delta. This value is
+    // required to report the high watermark (one offset past the last
+    // consumable offset).
+    auto next_offset = model::next_offset(manifest.get_last_offset());
```

Comment:
Wrapped this in a try-catch. I think that case shouldn't be possible since if we've truncated the entire manifest, presumably we've also already updated the offset_delta in the manifest and exited this loop above.